commit 711bd82a9d5551dd7d6e64ca7d17772aca32ba92 (refs/changes/72/4472/1)
Author: Hans Rosenfeld <hans.rosenfeld@joyent.com>
Date:   2018-07-03T17:50:48+02:00 (1 year, 3 months ago)
    
    OS-7061 bhyve: set MAP_NORESERVE when mmap'ing guest memory

diff --git a/usr/src/lib/libvmmapi/common/vmmapi.c b/usr/src/lib/libvmmapi/common/vmmapi.c
index 7ded3a82b5..6622a62d53 100644
--- a/usr/src/lib/libvmmapi/common/vmmapi.c
+++ b/usr/src/lib/libvmmapi/common/vmmapi.c
@@ -455,6 +455,9 @@ vm_setup_memory(struct vmctx *ctx, size_t memsize, enum vm_mmap_style vms)
 	 */
 	len = VM_MMAP_GUARD_SIZE + objsize + VM_MMAP_GUARD_SIZE;
 	flags = MAP_PRIVATE | MAP_ANON | MAP_NOCORE | MAP_ALIGNED_SUPER;
+#ifndef __FreeBSD__
+	flags |= MAP_NORESERVE;
+#endif
 	ptr = mmap(NULL, len, PROT_NONE, flags, -1, 0);
 	if (ptr == MAP_FAILED)
 		return (-1);
